analyzer/codechecker_analyzer/analyzers/clangsa/ctu_autodetection.py

```diff
@@ -171,6 +171,23 @@ def is_ctu_capable(self):
         return invoke_binary_checked(tool_path, ['-version'], self.environ) \
             is not False
 
+    @property
+    def extdef_mapping_can_read_pch(self):
+        """
+        Detects if the current clang-extdef-mapping can read PCH (.ast) files.
+        """
+
+        tool_path = self.mapping_tool_path
+
+        if not tool_path:
+            return False
+
+        help = invoke_binary_checked(tool_path, ['-help'], self.environ)
+        LOG.debug("output of clang-extdef-mapping\n %s" % help)
+        if help is False:
+            return False
+        return 'ast files' in help
```

Comment:
This is the content.
```
OVERVIEW:
This tool collects the USR name and location of external definitions in the source files (excluding headers).
Input can be either source files that are compiled with compile database or .ast files that are created from clang's -emit-ast option.

USAGE: clang-extdef-mapping [options] <source0> [... <sourceN>]

OPTIONS:

Generic Options:

  --help                      - Display available options (--help-hidden for more)
  --help-list                 - Display list of available options (--help-list-hidden for more)
  --version                   - Display the version of this program

clang-extdefmapgen options:

  --extra-arg=<string>        - Additional argument to append to the compiler command line
  --extra-arg-before=<string> - Additional argument to prepend to the compiler command line
  -p=<string>                 - Build path

-p <build-path> is used to read a compile command database.

        For example, it can be a CMake build directory in which a file named
        compile_commands.json exists (use -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
        CMake option to get this output). When no build path is specified,
        a search for compile_commands.json will be attempted through all
        parent paths of the first input file . See:
        https://clang.llvm.org/docs/HowToSetupToolingForLLVM.html for an
        example of setting up Clang Tooling on a source tree.

<source0> ... specify the paths of source files. These paths are
        looked up in the compile command database. If the path of a file is
        absolute, it needs to point into CMake's source tree. If the path is
        relative, the current working directory needs to be in the CMake
        source tree and the file must be in a subdirectory of the current
        working directory. "./" prefixes in the relative files will be
        automatically removed, but the rest of a relative path must be a
        suffix of a path in the compile command database.
```

> Is this two-word string quite reliable?
I think it is, these two words are included only in the new version that is capable of handling pch files.